monash-human-power/MHP_DAS
# Add MQTT Client class

## Description


This pull request adds an additional folder in data-acquisition-system/wireless_modules called middle_module which contains files that help the ESP32 connect to the WIFI and a client Class file which can be used to help connect, send and receive messages over the MQTT messaging protocol. This files is independent and so can be re-used by anyone wanting to use MQTT. 
The folder also contains a test file and a main.py which is still in the works to read sensor data for the middle wireless module.

## Screenshots

The test.py output (on the left) and the main.py output (on the right):
<img width="810" alt="Test main" src="https://user-images.githubusercontent.com/63642262/89255087-268bef00-d664-11ea-858f-2e66bf0b11af.png">


## Steps to Test

These are included in the README in the new folder, but I have it restated here again:
1. Ensure your ESP32 board is connected to the WIFI, to do this you can use the boot.py file and a locally created config.py and connect it (this would also In the process confirm that boot.py works)
2.  Open the test.py file on a IDE on the local computer, ensure the paho-mqtt package for python is installed
3. Place the main.py and MQTT_Class_client.py files onto the ESP32 board
4. Run the main.py file and without wasting too much time run the test.py file
5. You should see the test.py output messages sent from main.py on the console. This would confirm that both main.py and MQTT_Client_class.py works (main.py uses all the methods in the Client class)
